test/e2e/aws_test.go

```diff
@@ -81,53 +79,97 @@ func initRegion() error {
 
 var _ = Describe("AWS", func() {
 	var (
-		cluster kind.Cluster
-		client  *clientset.Clientset
+		kindCluster kind.Cluster
+		client      *clientset.Clientset
 	)
 	BeforeEach(func() {
 		fmt.Fprintf(GinkgoWriter, "running in AWS region: %s\n", region)
-		cluster.Setup()
-		cfg := cluster.RestConfig()
+		kindCluster = kind.Cluster{
+			Name: "capa-test-" + util.RandomString(6),
+		}
+		kindCluster.Setup()
+		cfg := kindCluster.RestConfig()
 		var err error
 		client, err = clientset.NewForConfig(cfg)
 		Expect(err).To(BeNil())
 	}, kindTimeout)
 
 	AfterEach(func() {
-		cluster.Teardown()
+		kindCluster.Teardown()
 	})
 
 	Describe("control plane node", func() {
 		It("should be running", func() {
-
 			sess := getSession()
 			accountID := getAccountID(sess)
 
 			createKeyPair(sess)
 			createIAMRoles(sess, accountID)
 
-			createNamespace(cluster.KubeClient())
+			namespace := "test-" + util.RandomString(6)
+			createNamespace(kindCluster.KubeClient(), namespace)
 
 			By("Creating a cluster")
+			clusterName := "test-" + util.RandomString(6)
+			fmt.Fprintf(GinkgoWriter, "Creating Cluster named %q\n", clusterName)
 			clusterapi := client.ClusterV1alpha1().Clusters(namespace)
-			_, err := clusterapi.Create(makeCluster())
+			_, err := clusterapi.Create(makeCluster(clusterName))
 			Expect(err).To(BeNil())
 
 			By("Creating a machine")
+			machineName := "cp-1"
+			fmt.Fprintf(GinkgoWriter, "Creating Machine named %q for Cluster %q\n", machineName, clusterName)
 			machineapi := client.ClusterV1alpha1().Machines(namespace)
-			_, err = machineapi.Create(makeMachine())
+			_, err = machineapi.Create(makeMachine(machineName, clusterName))
 			Expect(err).To(BeNil())
 
 			Eventually(
 				func() (*capa.AWSMachineProviderStatus, error) {
-					machine, err := machineapi.Get(controlPlaneName, metav1.GetOptions{})
+					machine, err := machineapi.Get(machineName, metav1.GetOptions{})
 					if err != nil {
 						return nil, err
 					}
+					if machine.Status.ProviderStatus == nil {
+						return &capa.AWSMachineProviderStatus{
+							InstanceState: &capa.InstanceStatePending,
+						}, nil
+					}
 					return capa.MachineStatusFromProviderStatus(machine.Status.ProviderStatus)
 				},
 				10*time.Minute, 15*time.Second,
 			).Should(beHealthy())
+
+			// TODO: Retrieve Cluster kubeconfig
+			// TODO: Deploy Addons
+			// TODO: Validate Node Ready
+			// TODO: Deploy additional Control Plane Nodes
+			// TODO: Deploy a MachineDeployment
+			// TODO: Scale MachineDeployment up
+			// TODO: Scale MachineDeployment down
+
+			// Nasty little hack to ensure that the Control Plane instance has come up
```

Comment:
Do you want to add an Eventually that checks for the control plane ready annotation, rather than hard coding a 5m sleep?

Reply:
Done, still testing the latest changes, though.